kedro-org/kedro-viz
# Upgrade to Node 18

## Description

Fixes #1811

This ticket is to update node from version 16 (which has [reached its end-of-life date](https://nodejs.org/en/blog/announcements/nodejs16-eol)) to version 18 (which has [entered the long-term support](https://nodejs.org/en/blog/announcements/v18-release-announce))

## Development notes

In order to get the app working with node 18, we also updated the following packages to new versions
- react-script: 5.0.1
- css-loader: 6.4.0
- postcss-loader: 8.1.1
- sass-loader: 14.1.1
- workerize-loader: 2.0.2
- webpack: 5.91.0


## QA notes
Steps to follow to test:
1. remove package-lock.json, and node-modules
2. run `npm install`
3. run ` make build` then `make run` and make sure the app is building properly

**Observation while testing it locally**
It feels like the stateful URL are not working as expected since I updated Node 18 and react-script.  Occasionally, clicking on a node fails to highlight it, or accessing it through the URL doesn't work as intended. Let me know if you see the same

## Checklist


gitpod link: https://gitpod.io/new/#github.com/kedro-org/kedro-viz/pull/1848
